include/deal.II/multigrid/mg_constrained_dofs.h

```diff
@@ -65,7 +65,19 @@ class MGConstrainedDoFs : public Subscriptor
   template <int dim, int spacedim>
   void initialize(const DoFHandler<dim,spacedim> &dof,
                   const typename FunctionMap<dim>::type &function_map,
-                  const ComponentMask &component_mask = ComponentMask());
+                  const ComponentMask &component_mask = ComponentMask()) DEAL_II_DEPRECATED;
+
+  /**
+   * Fill the internal data structures with information about interface and boundary dofs.
+   *
+   * This function internally calls the initialize() function above and the
+   * constrains degrees on the external boundary of the domain by calling
+   * MGTools::make_boundary_list() with the given second and third argument.
+   */
+  template <int dim, int spacedim>
+  void set_zero_boundary_dofs(const DoFHandler<dim,spacedim> &dof,
+                              const std::vector<types::boundary_id> &boundary_indicators,
```

Comment:
I would prefer if we would use a std::set for this argument because it avoids the need to search through the vector or document how exactly it needs to be sorted.